.woodpecker.yml

```diff
@@ -65,13 +68,16 @@ steps:
       cache_key: "jerboa-cache"
       region: us-east-1
       path-style: true
+      backend_operation_timeout: 60m
+      compression_level: 0
+      exit_code: true
       mount:
         - ".gradle"
     secrets:
       [MINIO_ENDPOINT, MINIO_WRITE_USER, MINIO_WRITE_PASSWORD, MINIO_BUCKET]
     when:
-      - path:
-          include: ["app/build.gradle.kts"]
```

Comment:
Could this be kept? I had added this, because I only want to save the new caches if something had changed.

Reply:
The problem is that anyone can change the file in a PR, and then potentially get access to secrets to mess with the cache. Maybe a better option would be a cronjob which automatically rebuilds the cache every night or so.

Reply:
Actually this wouldn't prevent that at all. Anyone can edit this workflow, set whatever event they want and trigger a pr that updates the caches

Reply:
I don't know how Dessalines Setup the secrets but you can set it up that only the main branch can access the secrets. That is probably a better solution see [this](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions#creating-secrets-for-an-environment)

Reply:
In Github you can probably do a "hack" where you can set a secret only available (in Environment) if the file hasn't changed and another environment that allows only access to the same secret if approved by a required reviewer.

In Woodpecker it doesn't seem possible to do something similar. https://woodpecker-ci.org/docs/usage/secrets#use-in-pull-requests-events

Reply:
Thats true. Ive changed it back now.

Reply:
If this is deemed too much of a security hazard, the caching can be removed, we don't gain that much from it. The biggest improvement was already made by setting the .gradle as ENV. This allowed it to reuse the gradle cache between steps

Reply:
Its mostly a theoretical problem, at least so far it hasnt been abused in practice. And we would probably notice if someone opened a malicious PR.

Reply:
Seems it failed to download a depency, network issues prob, try again by pushing a commit